src/lib/kdb/kdb_default.c

```diff
@@ -538,3 +538,52 @@ krb5_def_fetch_mkey_list(krb5_context        context,
         krb5_dbe_free_key_list(context, mkey_list_head);
     return retval;
 }
+
+krb5_error_code
+krb5_db_def_rename_principal(krb5_context kcontext,
+                             krb5_const_principal source,
+                             krb5_const_principal target)
+{
+    krb5_db_entry *kdb = NULL;
+    krb5_principal new, old;
+
+    krb5_error_code ret;
+
+    if (source == NULL || target == NULL)
+        return EINVAL;
+
+    ret = krb5_db_get_principal(kcontext, source,
+                                KRB5_KDB_FLAG_ALIAS_OK, &kdb);
+    if (ret)
+        goto cleanup;
+
+    /*
+     * Store salt values explicitly so that they don't depend on the principal
+     * name.
+     */
+    ret = krb5_dbe_specialize_salt(kcontext, kdb);
+    if (ret)
+        goto cleanup;
+
+    ret = krb5_copy_principal(kcontext, target, &new);
```

Comment:
We don't need to make this copy; we can just assign kdb->princ = target temporarily.